internal/painter/gl/painter.go

```diff
@@ -36,6 +36,8 @@ type Painter interface {
 // If it is a master painter it will also initialise OpenGL
 func NewPainter(c fyne.Canvas, ctx driver.WithContext) Painter {
 	p := &painter{canvas: c, contextProvider: ctx}
+	p.blurKernels = make(map[float32][]float32)
```

Comment:
Is there a reason not to cache the kernel on the `canvas.Blur` itself? With a map like this we need a clean task / expiry too, and it seems that caching it on the blur object would avoid the overhead of a map lookup (which, granted, is small). You can have
```go
package "canvas"
type Blur struct {
    // ... public fields
    
    cachedKernelRadius float32
    cachedKernel []float32
}
```

Then you can know whether the kernel needs to rebuilt for a new radius (if the .Radius property on the blur has been updated).

Reply:
For all the ones the theme applies to dialog backgrounds, yes. But user code could do something like animate a blur over a range of radii, causing a potentially unbounded number of entries into the blurKernels cache. We can keep it a map, but we'll have to make sure we have a system to delete from it, like all caches.

Reply:
worth noting that [this issue](https://github.com/fyne-io/fyne/issues/5486) would apply here as well - but I had a draft PR to address this (which I really should pick up again sometime) and can add the `blurKernel` cache to this new system as well.

Reply:
Bear in mind the kernel cache is a OpenGL requirement not all renderers. 

Can you expand on why caching a rendering lookup table on the object is clearer?

Also I'm unsure how a private field on a struct in a different package is even something the renderer code can reference... I feel I'm missing something.

Reply:
I'm not getting why we would choose to re-compute every kernel for every frame over using some memory to cache them...

Reply:
We can use memory to cache them but we **must** add the cache clean task then, so that we don't retain that memory indefinitely for every blur radius ever asked for by the app.